### PR TITLE
Fix issues for az aks kollect on private clusters

### LINKED ISSUE #1679: AKS Private-Cluster fails with Nonetype stackTrace via Jumpbox
### Extension name (the extension in question)

* `aks-preview`

### Description of issue (in as much detail as possible)

_Before I detail_: This is placeholder issue for transparency, plus to log and correctly identify the `aks private-cluster` behavior to rectify the code. If this is Network \ Storage issue then its worth to identify that as well and involve right folks to identify correct fix for it.

**Steps to Reproduce**:

* Create az-aks Private cluster: https://docs.microsoft.com/en-us/azure/aks/private-clusters
	* Obviously from the local dev box the `privatelink` will not be visible so pursue the `jumpbox` vm approach. i.e. create vm in same subnet as aks private cluster.
* After creating jumpbox vm > install and run `az cli` and add > `aks=preview` extension.
* Run `AKS Kollect` command for the private cluster from the jumpbox, the [`aks_kollect` call ](https://github.com/Azure/azure-cli-extensions/blob/master/src/aks-preview/azext_aks_preview/custom.py#L1321) **fails to get any result for line** `1332` -- `client.get` call in `custom.py`  gives empty result for the `private cluster` hence the error like below where it tries toaccess `fqdn` (fullyqualified domain name) but it is `None` or refer to screenshot. This might very well be connected to a portal issue where for the `private ak cluster` the **Properties** tab shows empty API Server address.

* This issue needs deeper investigation and then right solution needs to be implemented. If you know much about how `Private-Cluster` behave please feel free to reachout.
    * Further: if its the case where >> `kollect` command do not support `private-cluster` we should update docs to represent that.

Please feel free to assign back to me, but this is added for mere transparency and to get right folks eyes for comments.

Just as fyi: @Jen7714 and @yunjchoi 

```
  File "/root/.azure/cliextensions/aks-preview/azext_aks_preview/custom.py", line 1485, in aks_kollect
    normalized_fqdn = mc.fqdn.replace('.', '-')
AttributeError: 'NoneType' object has no attribute 'replace'
```

image

Sample command: 

```
az aks kollect 
-g MyResourceGroup
-n **Private_ManagedCluster**
--storage-account MyStorageAccount
--sas-token "MySasToken"
```

### PR DESCRIPTION
**Pull request by feiskyer**
Fixes #1679

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -1478,7 +1478,8 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
         os.remove(temp_yaml_path)
 
     print()
-    normalized_fqdn = mc.fqdn.replace('.', '-')
+    fqdn = mc.fqdn if mc.fqdn is not None else mc.private_fqdn
```
Comment (Tatsinnit) [inline src/aks-preview/azext_aks_preview/custom.py:1481]: 👍  LGTM, nice, quick question: do we have any `msdn` doc which list these `objects`

**Comment by yonzhan** [comment]
add to S169

File: src/aks-preview/azext_aks_preview/custom.py (hunk 1 above)
Comment (feiskyer) [reply src/aks-preview/azext_aks_preview/custom.py:1481]: those properties are coming from AKS rest API, e.g. https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/get

Comment (Tatsinnit) [reply src/aks-preview/azext_aks_preview/custom.py:1481]: +1 /LGTM

File: src/aks-preview/azext_aks_preview/_helpers.py [hunk 2]
```diff
@@ -88,4 +88,6 @@ def _trim_fqdn_name_containing_hcp(normalized_fqdn: str) -> str:
     attached
     """
     storage_name_without_hcp, _, _ = normalized_fqdn.partition('-hcp-')
+    if len(storage_name_without_hcp) > 63:
```
Comment (yungezz) [inline src/aks-preview/azext_aks_preview/_helpers.py:91]: how's the 63 coming from, just curious

Comment (feiskyer) [reply src/aks-preview/azext_aks_preview/_helpers.py:91]: from the docs here: https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names.

Comment (arrownj) [reply src/aks-preview/azext_aks_preview/_helpers.py:91]: Can we define a constant for it and add the link to the code comment ?

Comment (feiskyer) [reply src/aks-preview/azext_aks_preview/_helpers.py]: added.

**Comment by feiskyer** [comment]
@arrownj  thanks, should be fixed now.

**Comment by feiskyer** [comment]
@arrownj could we get this merged now?

**Pull request merged**